tests/typescript/src/field/different_user_input.ts

```diff
@@ -0,0 +1,75 @@
+/**
+ * Test: Should allow a subclass of Field to have different setValue and
+ * constructor input type than the type that is stored.
+ */
+import { Field, FieldValidator, fieldRegistry } from 'blockly-test/core';
+
+interface Cell {
+    cellId: string;
+}
+
+interface CellGroup {
+    cells: Cell[];
+}
+
+type FieldMitosisValidator = FieldValidator<CellGroup>;
+
+class FieldMitosis extends Field<CellGroup>{
+    constructor(cell: Cell, validator: FieldMitosisValidator) {
+        super(Field.SKIP_SETUP);
+
+        this.setValue(cell);
+        this.setValidator(validator);
+    }
+
+    // Overwritten Field classes.
+
+    protected doClassValidation_(newCell?: unknown): CellGroup | null {
+        if (!this.isCell(newCell)) return null;
+
+        const cellGroup = this.getValue() ?? { cells: [] };
+        cellGroup.cells.push(newCell);
+        return cellGroup;
+    }
+
+    // Example-specific classes.
+
+    private isCell(maybeCell: unknown): maybeCell is Cell {
+        if (!maybeCell) return false;
+        const couldBeCell = maybeCell as { [key: string]: unknown };
+        return 'cellId' in couldBeCell && typeof couldBeCell.cellId === 'string';
```

Comment:
Whoops! Perhaps my comments above mistakenly mentioning `classes` instead of `methods` may have contributed to this misunderstanding. 

Regarding using an `interface` vs a `class`, I don't think a class really makes sense here for either `Cell` or `CellGroup`.  Although there is the extra `isCell` function, we don't have to invoke a `new Cell` every time one needs to be created and we don't need to setup a constructor and methods for the `Cell`/`CellGroup` objects to create them.  Also, classes are ideal when you have complex state that needs to be maintained and/or when you have methods/values that are valuable for a subclass to inherit as they are. In this case, the purpose of the example is that `FieldMitosis` manages the state of these values. If we wanted to use a class, there should really be one class that encompasses `Cell` and `CellGroup` and handles methods like `doMitosis`, and that would invalidate this example.

Part of this may be personal preference, though, as I tend to avoid classes when possible. Classes are great in JavaScript when there's a bunch of uncertainty since everything is in one spot and you already have to keep everything involved in mind since you can't always trust that things are what they say they are. In TypeScript, though, using functions + many interfaces is great since you only have to keep in mind what the state is at that one point in time. If state should change, just invoke a function to transform the state to what it should be and have the resulting type reflect that. Makes functions super simple, too, since they just worry about the state coming in and how to produce the state going out.